sdk-actors/src/main/java/io/dapr/actors/runtime/DaprStateAsyncProvider.java

```diff
@@ -68,6 +68,19 @@
           return Mono.empty();
         }
 
+        if (!this.isStateSerializerDefault) {
+          if (s.length == 0) {
+            return Mono.empty();
+          }
+          s = OBJECT_MAPPER.readValue(s, byte[].class);
+          if (type.getType().equals(String.class)) {
+            byte[] out = new byte[s.length + 2];
+            out[0] = '"';

```

Comment:
This does not cover the scenarios where the string contains a quote already since it would not be escaped.

Reply:
Quotes will be added in the serialization process. Give me please an example I'll add it in tests

Reply:
`my "string" with quotes`

Reply:
![image](https://user-images.githubusercontent.com/15830587/129274564-876ef740-83bc-42e2-81ca-e6308eaba11c.png)
It'll be quoted in the serialization process.

Reply:
I add two cases for quoted strings. As I saw in debug additional quotes had been escaped with serializer.

Reply:
I am still concerned about this logic. This should not be needed. I will need more time to try this PR on my desktop first.